test/helpers/rule-test-harness.js

```diff
@@ -1,176 +1,13 @@
-          expect(actual[0]).to.deep.equal(badItem.fatal);
-        } else {
-          expectedResults = expectedResults.map(parseResult);
-
-          expect(actual).to.deep.equal(expectedResults);
-        }
-      });
-
-      if (!skipDisabledTests) {
-        testMethod(`passes with \`${template}\` when rule is disabled`, function() {
-          config = false;
-          meta = parseMeta(badItem);
-          let actual = verify(template);
-
-          expect(actual).to.deep.equal([]);
-        });
-
-        testMethod(
-          `passes with \`${template}\` when disabled via inline comment - single rule`,
-          function() {
-            meta = parseMeta(badItem);
-            let actual = verify(DISABLE_ONE + '\n' + template);
-
-            expect(actual).to.deep.equal([]);
-          }
-        );
-
-        testMethod(
-          `passes with \`${template}\` when disabled via inline comment - all rules`,
-          function() {
-            meta = parseMeta(badItem);
-            let actual = verify(DISABLE_ALL + '\n' + template);
-
-            expect(actual).to.deep.equal([]);
-          }
-        );
-      }
-    });
-
-    options.good.forEach(function(item) {
-      let template = typeof item === 'object' ? item.template : item;
-      let testMethod = typeof item === 'object' && item.focus ? it.only : it;
-
-      testMethod(`passes when given \`${template}\``, function() {
-        meta = parseMeta(item);
-        let actual;
-
-        if (typeof item === 'string') {
-          actual = verify(item);
-        } else {
-          if (item.config !== undefined) {
-            config = item.config;
-          }
-
-          actual = verify(template);
-        }
-
-        expect(actual).to.deep.equal([]);
-      });
-    });
-
-    let error = options.error || [];
-    error.forEach(item => {
-      let template = item.template;
-      let testMethod = item.focus ? it.only : it;
-
-      if (item.config !== undefined) {
-        config = item.config;
-      }
-
-      let friendlyConfig = JSON.stringify(config);
-
-      let _config = config;
-
-      testMethod(`errors when given \`${template}\` with config \`${friendlyConfig}\``, function() {
-        let expectedResults = item.results || [item.result];
-        meta = parseMeta(item);
-        expectedResults = expectedResults.map(parseResult);
-
-        config = _config;
-
-        let actual = verify(template);
-
-        for (let i = 0; i < actual.length; i++) {
-          if (actual[i].fatal) {
-            delete expectedResults[i].rule;
-            delete actual[i].source;
-
-            expect(actual[i].message).to.contain(expectedResults[i].message);
-
-            delete actual[i].message;
-            delete expectedResults[i].message;
-          }
-        }
-
-        expect(actual).to.deep.equal(expectedResults);
-      });
-    });
-  });
+  return generateRuleTests(
+    Object.assign(options, {
```

Comment:
To avoid mutating the passed in `options` let's do:

```suggestion
    Object.assign({}, options, {
```